src/App.tsx

```diff
@@ -1,12 +1,61 @@
 import React from 'react';
 import './App.scss';
+import { GoodsList } from './components/GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type Props = {};
+
+type State = {
+  visibleGoods: Good[]
+};
+
+class App extends React.Component<Props, State> {
+  state: State = {
+    visibleGoods: [],
+  };
+
+  handlerClick = (getGoods: Promise<Good[]>) => {
+    getGoods.then(goods => this.setState({
```

Comment:
you should call function here, not in parameters

```suggestion
    getGoods().then(goods => this.setState({
```